src/olympia/yara/models.py

```diff
@@ -0,0 +1,26 @@
+from django.db import models
+from django_extensions.db.fields.json import JSONField
+
+from olympia.amo.models import ModelBase
+from olympia.files.models import FileUpload
+
+
+class YaraResult(ModelBase):
+    upload = models.OneToOneField(FileUpload,
+                                  related_name='yara_results',
+                                  on_delete=models.CASCADE)
```

Comment:
Note: `FileUpload` instances are _supposed_ to be a short term thing before the addon/version has been created, and ignored afterwards.  There's a cron job that deletes old `FileUpload`s, which would (because `on_delete=models.CASCADE`) end up deleting the `YaraResult` too (though I think the cron might be broken).  So you possibly don't want `CASCADE` for this fk.

Reply:
It depends on how long you intend for `YaraResult`s to be kept around, of course.

Reply:
Ah, correct. I believe removing a `FileUpload` should not remove the `YaraResult` because this relation is only useful during the submission process, until we can attach a `Version` to the `YaraResult`. So I changed (1) the column to be nullable and (2) the FK constraint to `SET NULL` on delete.

Reply:
I tried to delete a `FileUpload` and [everything](https://github.com/mozilla/addons-server/compare/master...willdurand:yara) continues to work as expected.